### PR TITLE
Batch data issue#138

### PR DESCRIPTION
**Pull request by bhawna333**

### CODE REVIEW

**Comment by mstockerl** [comment]
Thanks for making the changes! 👍 

As we have the logic to display the batches you can apply for or will be able to apply for in the future on several places, I wonder whether it makes sense to have an own component for it. 
What do you think?

**Comment by bhawna333** [comment]
> Thanks for making the changes! 
> 
> As we have the logic to display the batches you can apply for or will be able to apply for in the future on several places, I wonder whether it makes sense to have an own component for it.
> What do you think?

Yup it makes sense since we are reusing the code now at more than 2 places. I'll try to make it into a component then and see how it works out

**Comment by bhawna333** [comment]
Also addressed the issue #137 regarding text change for currently open applications in this pull request

File: src/components/shared/batch-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+class BatchDetails extends React.Component<BatchDetailsProps> {
+  constructor(props: BatchDetailsProps) {
+    super(props);
+    this.state = { batchDetails: [] };
+  }
+
+  render() {
+    let displayAllBatches = this.state.batchDetails.map(
```
Comment (mstockerl) [inline src/components/shared/batch-details/index.tsx]: I'd do this in the if-else statement below.

File: src/components/shared/batch-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+              {`(Application phase: ${getBatchDate(
+                batch.appStartDate
+              )} to ${getBatchDate(batch.appEndDate)})`}
+              <br />
+            </span>
+          );
+      }.bind(this)
+    );
+
+    let displayCurrentBatches = this.state.batchDetails.map(
```
Comment (mstockerl) [inline src/components/shared/batch-details/index.tsx]: I'd also put this into the if-else statement below

File: src/components/shared/batch-details/index.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { getFirebase } from '../../../firebase/firebase';
+
+interface BatchDetailsProps {
+  isCurrentOpenApplications: boolean;
+}
+
+function getBatchDate(batchDate) {
```
[the remark this answers is not in the document]
Comment (bhawna333) [reply src/components/shared/batch-details/index.tsx]: Not necessary, its just a helper function. I can put it inside, but I don't know of the advantages of putting it inside.

**Pull request merged**